core/heap.c

```diff
@@ -895,7 +896,19 @@ vmm_heap_unit_exit(vm_heap_t *vmh)
     /* In case there are no tombstones we can just free the unit and
      * that is what we'll do, otherwise it will stay up forever.
      */
-    if (vmh->num_free_blocks == vmh->num_blocks) {
+    bool free_heap = vmh->num_free_blocks == vmh->num_blocks;
+#ifdef UNIX
+    /* On unix there's no fear of leftover tombstones, and as long as we're
+     * doing a detach we can be sure our stack is not actually in the heap.
+     */
+    if (doing_detach) {
+      byte *sp;
+      GET_STACK_PTR(sp);
```

Comment:
Put this all inside a DODEBUG as it's only for the assert

Reply:
Done